src/main/java/codeu/controller/LoginServlet.java

```diff
@@ -54,32 +54,37 @@ void setUserStore(UserStore userStore) {
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException, ServletException {
+      throws IOException, ServletException{
     request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
   }
 
-  /**
-   * This function fires when a user submits the login form. It gets the username from the submitted
-   * form data, and then adds it to the session so we know the user is logged in.
-   */
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response)
-      throws IOException, ServletException {
-    String username = request.getParameter("username");
+/**
+  * This function fires when a user submits the login form. It gets the username and password from
+  * the submitted form data, checks that they're valid, and either adds the user to the session
+  * so we know the user is logged in or shows an error to the user.
+  */
 
-    if (!username.matches("[\\w*\\s*]*")) {
-      request.setAttribute("error", "Please enter only letters, numbers, and spaces.");
-      request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
-      return;
-    }
+ @Override
+ public void doPost(HttpServletRequest request, HttpServletResponse response)
```

Comment:
Looks like the changes to this method are causing a few failures in LoginServletTest